core/unit_test/sycl/TestSYCL_InterOp_StreamsMultiGPU.cpp

```diff
@@ -0,0 +1,78 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <TestSYCL_Category.hpp>
+#include <TestMultiGPU.hpp>
+
+namespace {
+
+std::array<TEST_EXECSPACE, 2> get_execution_spaces() {
+  std::vector<sycl::device> gpu_devices =
+      sycl::device::get_devices(sycl::info::device_type::gpu);
+  auto exception_handler = [](sycl::exception_list exceptions) {
+    bool asynchronous_error = false;
+    for (std::exception_ptr const &e : exceptions) {
+      try {
+        std::rethrow_exception(e);
+      } catch (sycl::exception const &e) {
+        std::cerr << e.what() << '\n';
+        asynchronous_error = true;
+      }
+    }
+    if (asynchronous_error)
+      Kokkos::Impl::throw_runtime_exception(
+          "There was an asynchronous SYCL error!\n");
+  };
+
+  TEST_EXECSPACE exec0(sycl::queue{gpu_devices.front(), exception_handler,
+                                   sycl::property::queue::in_order()});
+  TEST_EXECSPACE exec1(sycl::queue{gpu_devices.back(), exception_handler,
+                                   sycl::property::queue::in_order()});
+
+  return {exec0, exec1};
+}
+
+TEST(sycl_multi_gpu, managed_views) {
+  std::array<TEST_EXECSPACE, 2> execs = get_execution_spaces();
+
+  Kokkos::View<int *, TEST_EXECSPACE> view0(Kokkos::view_alloc("v0", execs[0]),
+                                            100);
+  Kokkos::View<int *, TEST_EXECSPACE> view(Kokkos::view_alloc("v", execs[1]),
+                                           100);
+
+  test_policies(execs[0], view0, execs[1], view);
+}
+
+TEST(sycl_multi_gpu, unmanaged_views) {
+  std::array<TEST_EXECSPACE, 2> execs = get_execution_spaces();
+
+  int *p0 = sycl::malloc_device<int>(100, execs[0].sycl_queue());
```

Comment:
Would you remind me why we need to test both for managed and unmanaged views?

Reply:
So we test interoperability with GPU memory pointers as well as allocations going through managed Views.
If only the one with managed Views fails, we have a good idea that something in the View allocation is problematic.

Reply:
I am not blocking but I feel like it is not worth the complexity of having both.

Reply:
In any case please update the description and mention the refactor to reuse the tests for CUDA multi-GPUs

Reply:
Done.